prediction_market_agent/agents/autogen_general_agent/farcaster_agent.py

```diff
@@ -0,0 +1,155 @@
+import datetime
+from enum import Enum
+from string import Template
+from typing import Any, Dict, Optional
+
+import autogen
+from autogen import AssistantAgent, UserProxyAgent
+from autogen.cache import Cache
+from loguru import logger
+from prediction_market_agent_tooling.markets.agent_market import FilterBy, SortBy
+from prediction_market_agent_tooling.markets.omen.omen_subgraph_handler import (
+    OmenSubgraphHandler,
+)
+from prediction_market_agent_tooling.tools.utils import utcnow
+
+from prediction_market_agent.agents.autogen_general_agent.prompts import (
+    CRITIC_PROMPT,
+    INFLUENCER_PROMPT,
+)
+from prediction_market_agent.utils import APIKeys
+
+# We have an influencer and a critic. Based on https://microsoft.github.io/autogen/docs/notebooks/agentchat_nestedchat/.
+
+
+class AutogenAgentType(str, Enum):
+    WRITER = "writer"
+    CRITIC = "critic"
+    USER = "user"
+
+
+def reflection_message(
+    recipient: UserProxyAgent,
+    messages: list[Dict[Any, Any]],
+    sender: AssistantAgent,
+    config: Optional[Any] = None,
```

Comment:
Why is config `Any` and then not used?

Reply:
These args are required by autogen, and Any is there due to mypy complaints.